heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java

```diff
@@ -39,12 +40,105 @@
 import com.twitter.heron.spi.utils.ShellUtils;
 
 public final class SchedulerUtils {
-  public static final int PORTS_REQUIRED_FOR_EXECUTOR = 9;
+  public static final int PORTS_REQUIRED_FOR_EXECUTOR = ExecutorPorts.getRequiredNumOfPorts();
   public static final int PORTS_REQUIRED_FOR_SCHEDULER = 1;
   public static final String SCHEDULER_COMMAND_LINE_PROPERTIES_OVERRIDE_OPTION = "P";
 
   private static final Logger LOG = Logger.getLogger(SchedulerUtils.class.getName());
 
+  /**
+   * Enum that defines the type of ports that an heron executor needs
+   */
+  public static class ExecutorPorts extends HashMap<ExecutorPorts.ExecutorPortNames, Object> {
```

Comment:
Why is `Object` used here instead of `Integer`?

Reply:
This is done to be generic so that some options in the future like ports for remote debugging of the jvm can be a list of integers as well:

https://github.com/twitter/heron/pull/2484

Reply:
I think this added flexibility makes use of this class complex, requiring casting to the correct type. The object could be anything, a `String`, a `Topology`. The user will need knowledge of all references before any modification can be made.

Reply:
I agree that ports should be Integers. Using `Object` for flexibility is almost always an anti-pattern. I would suggest using integer and if we find later that there's a requirement for multiple port bound to a single name (which seems like something to avoid IMO), then we would consider adding support for lists of integers.